### DISCUSSION TITLE
How to get current dirty data size？

### DISCUSSION THREAD

**Discussion by sunetic** [openzfs/zfs#9620] [Q&A] [answered]
### System information

Type | Version/Name
 --- | --- 
Distribution Name	| Redhat 
Distribution Version	| 7.4
Linux Kernel	| 3.10.0-693.el7.x86_64
Architecture	| x86
ZFS Version	| 0.6.5
SPL Version	| 0.6.5

### Describe the problem you're observing
I had set zfs_dirty_data_max but how to check the usage to adjust this parameter?

**Comment by ahrens** [comment]
This is tracked in `dsl_pool_t:dp_dirty_total`.  AFAIK it is not available in userland.

**Comment by richardelling** [comment, marked as answer]
For telemetry services, that value is too volatile to be useful. We do track the contents of `/proc/spl/kstat/zfs/dmu_tx` in both `node_exporter` and `telegraf`  The data there can be useful for monitoring delays due to the write throttle and I've got a dashboard or two that show examples of how to look at that data.

**Comment by sunetic** [comment]
@richardelling where could I found the two dashboard?thanks.

**Comment by aebch** [comment]
@richardelling, can you share the Dashboards?

**Comment by JonathonFS** [comment]
The OpenZFS wiki seems to indicate that you can track dirty data by reviewing the **ndirty** column in /proc/spl/kstat/zfs/{poolname}/txgs. I've been watching mine, and it definitely spikes up and down when I restart a VM workload. Saw it exceed my **zfs_dirty_data_max** by ~1.5 MB once. Otherwise it always stays below.

https://openzfs.org/wiki/ZFS_on_high_latency_devices
